src/VisualStudio/CSharp/Impl/CodeModel/CSharpCodeModelService.cs

```diff
@@ -1140,7 +1140,8 @@ private EnvDTE.vsCMAccess GetDefaultAccessibility(SyntaxNode node)
                 }
             }
 
-            if (node is AccessorDeclarationSyntax)
+            if (node is AccessorDeclarationSyntax || 
+                node is ArrowExpressionClauseSyntax)
             {
                 return GetAccess(node.FirstAncestorOrSelf<BasePropertyDeclarationSyntax>());
```

Comment:
no

changed the input code to
`public class Bar
{
    int a;

    public int A
    {
        get
        { return a; }
        
        $$private set {}
    }

    public string WindowsUserID => ""Domain""; 
}`

and the test to
`       [ConditionalWpfFact(typeof(x86))]
        [Trait(Traits.Feature, Traits.Features.CodeModel)]
        public void Accessor2()
        {
            CodeClass testObject = GetCodeClass("Bar");

            var l = from p in testObject.Members.OfType<CodeProperty>() where vsCMAccess.vsCMAccessPublic == p.Access && p.Setter != null && !p.Setter.IsShared && vsCMAccess.vsCMAccessPrivate == p.Setter.Access select p;
            var z = l.ToList<CodeProperty>();
            Assert.Equal(1, z.Count);
            string s = z[0].Name;
            Assert.Equal(s, "A");
        }`

which is looking for a public property with a private setter and it finds Property A as expected